From 37afc9f2d2f8de4bb038e23f85334dcd7c876e48 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Thu, 24 Nov 2016 17:56:58 +0200
Subject: [PATCH 1152/2241] tools: nas: add support for Armada-70x0/80x0 in
 nas_init script

- Add support for Armada-70x0/80x0
- Configure irq affinity for Armada-70x0/80x0

Change-Id: Ib99ea04309161f55ca7f6a557e0d689368c10be8
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34067
Reviewed-by: Ofer Heifetz <oferh@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 tools/nas/nas_init.sh | 39 ++++++++++++++++++++++++++++++++++++---
 1 file changed, 36 insertions(+), 3 deletions(-)

diff --git a/tools/nas/nas_init.sh b/tools/nas/nas_init.sh
index fa9d432..59750fa 100755
--- a/tools/nas/nas_init.sh
+++ b/tools/nas/nas_init.sh
@@ -103,7 +103,7 @@ do
 		;;
 	s)	PLATFORM=$OPTARG
 		case "$OPTARG" in
-			a380|a385|a388|a37xx|none) ;;
+			a380|a385|a388|a37xx|a8040|a7040|none) ;;
 			*) do_error "-${flag}: wrong option ${OPTARG}" ;;
 		esac
 		;;
@@ -132,7 +132,7 @@ do
 		fi
 		;;
 	*)	echo "Usage: $0"
-		echo "		-s <a380|a385|a388|a37xx|none>: platform used with nas_init"
+		echo "		-s <a380|a385|a388|a37xx|a8040|a7040|none>: platform used with nas_init"
 		echo "		-f <ext4|xfs|btrfs|fat32>: file system type ext4, xfs, btrfs or fat32"
 		echo "		-t <sd|rd0|rd1|rd5|rd6>: drive topology"
 		echo "		-n <num>: partition number to be mounted"
@@ -153,7 +153,7 @@ done
 
 # verify supporting arch
 case "$PLATFORM" in
-    a380|a385|a388|a37xx|none)	;;
+    a380|a385|a388|a37xx|a8040|a7040|none)	;;
     *)	do_error "Platform ${PLATFORM} unsupported, please pass valid -s option" ;;
 esac
 
@@ -785,6 +785,39 @@ elif [ "$PLATFORM" == "a37xx" ]; then
 
 	set +o verbose
 	echo -ne "[Done]\n"
+elif [ "$PLATFORM" == "a8040" ]; then
+	set -o verbose
+
+	# XOR Engines for AP
+	echo 1 > /proc/irq/67/smp_affinity
+	echo 2 > /proc/irq/68/smp_affinity
+	echo 4 > /proc/irq/69/smp_affinity
+	echo 8 > /proc/irq/70/smp_affinity
+
+	# SATA
+	echo 2 > /proc/irq/41/smp_affinity
+
+	# To use CP XOR engines, need to parse interrupt map and
+	# read interrupt numbers assigned to XOR engine (ICU driver
+	# dynamically assigns interrupts for CP interfaces)"
+
+	set +o verbose
+	echo -ne "[Done]\n"
+elif [ "$PLATFORM" == "a7040" ]; then
+	set -o verbose
+
+	# XOR Engines for AP
+	echo 1 > /proc/irq/123/smp_affinity
+	echo 2 > /proc/irq/124/smp_affinity
+	echo 4 > /proc/irq/125/smp_affinity
+	echo 8 > /proc/irq/126/smp_affinity
+
+	# To use CP XOR engines, need to parse interrupt map and
+	# read interrupt numbers assigned to XOR engine (ICU driver
+	# dynamically assigns interrupts for CP interfaces)"
+
+	set +o verbose
+	echo -ne "[Done]\n"
 else
 	echo -ne "[Skip]\n"
 fi
-- 
2.7.4

